src/modules/FtpServer.rb

```diff
@@ -539,36 +541,29 @@ def WriteFirewallSettings
       port_range = ""
       active_port = ""
 
-      if SuSEFirewall.IsStarted
-        if Ops.get(@EDIT_SETTINGS, "PassiveMode") == "YES"
-          port_range = Ops.add(
-            Ops.add(Ops.get(@EDIT_SETTINGS, "PasMinPort"), ":"),
-            Ops.get(@EDIT_SETTINGS, "PasMaxPort")
-          )
-        else
-          active_port = PortAliases.IsKnownPortName("ftp-data") ? "ftp-data" : "20"
-        end
-
-        suse_config = {
-          "tcp_ports" => [
-            PortAliases.IsKnownPortName("ftp") ? "ftp" : "21",
-            active_port != "" ? active_port : port_range
-          ]
-        }
+      return true if !firewalld.running?
 
-        if @vsftpd_edit
-          return SuSEFirewallServices.SetNeededPortsAndProtocols(
-            "service:vsftpd",
-            suse_config
-          )
-        else
-          return SuSEFirewallServices.SetNeededPortsAndProtocols(
-            "service:pure-ftpd",
-            suse_config
-          )
-        end
+      if Ops.get(@EDIT_SETTINGS, "PassiveMode") == "YES"
+        port_range = Ops.add(
+          Ops.add(Ops.get(@EDIT_SETTINGS, "PasMinPort"), ":"),
+          Ops.get(@EDIT_SETTINGS, "PasMaxPort")
```

Comment:
can it be nil? if yes, then result can be a bit strange in this add

Reply:
It shouldn't be `nil`. When it is left blank in the Input then it its filled with `1024`.